### DISCUSSION TITLE
About Python 2 support

### DISCUSSION THREAD

**Discussion by echoix** [OSGeo/grass#2055] [General]
I'm exploring the Python code base and trying to resolve a first issue for GRASS, but I have difficulty understanding some things. 

One of the most useful questions to answer is whether Python 2 support is officially dropped for GRASS 8 and/or 7.8.6.
I see, for example in the g.extension code, that the code base uses the six package to have a Python 3 code simulating the behaviour of Python 2 in a Python 3 code. I also see some release notes mentioning Python 3 support, or commit messages applying changes for Python 3, and even some new changes that aren't Python 2 compatible. But most importantly, when reading the discussions on the PRs and mailing lists about using Black, like https://github.com/OSGeo/grass/pull/1382, I understand that the generated formatting is supposed to be Python3 only. On Black's website and discussions, complete Python 2 support (and codepaths) will be removed in the first release in January https://black.readthedocs.io/en/stable/faq.html#does-black-support-python-2. So, dropping Python 2 from all code is inevitable, and is already done in major Python packages (numpy, matplotlib) for a long time.

I'm trying to investigate the issue in https://github.com/OSGeo/grass/issues/2025. Keeping Python 2 compatibility makes it harder to find the best fix while using common design patterns. But I see that commit 4705cc5b422cbd63abe6c8da0d917ebe957395d5 straight forwardly breaks Python 2 compatibility by using f-strings. 

So here it is, is dropping Python 2 from code official now?

**Comment by petrasovaa** [comment]
GRASS 8 (soon to be released) is Python 3 only (3.6+ specifically, but 3.6 is EOL anyway). That said, there was not much effort to explicitly remove Python2 support. 
With GRASS 7.8 it gets more complicated, the idea was to keep Python 2 compatibility, but practically it would be difficult to do that for the GUI. So I think libraries and modules may still work with python 2, but probably not GUI. Not sure if someone tested it recently. 
To summarize, I would just go with Python 3.7+.
